### ISSUE TITLE
RFE: Client Setting for Sensor Defaults

### ISSUE THREAD

**Issue by Saklad5** [MegaMek/megamek#5868]
Almost all units come with a set of sensors to choose from, with different strengths and weaknesses. Right now, MegaMek always defaults to using a unit's active probe if installed, and radar if not.

Radar is generally a poor choice on military units, especially next to options like magscan and infrared. As a result, I often start each fight tediously tweaking the sensor setting of each unit as I deploy them.

Rather than simply tweaking the default order to match my own preferences, I propose adding a client setting consisting of an ordered list of all sensors known to MegaMek. Each unit can then default to the first sensor it has on the list when deployed.

**Comment by SimonLandmine** [comment]
As an additional option, in some circumstances, it might be useful to be able to set the 'sensor in use' in the Unit Configuration window in the Lobby.
However, having a Client Setting (as requested above, in a similar choice to weapons display order) for 'Preferred Sensors' would potentially be more useful, especially with larger forces, avoiding having to configure each in turn.

**Comment by Saklad5** [comment]
The active sensors may already be adjusted in “Round Zero”, as you deploy units.

**Comment by SimonLandmine** [comment]
Yes, but configuring in the lobby would raise the potential for pre-saving sensor selection as a MUL along with other configuration. Also, I was noting one of the requests from the Discord. :-)

**Comment by Saklad5** [comment]
Ah, that does make sense.

**Comment by repligator** [comment]
See also https://github.com/MegaMek/mekhq/issues/4044

**Comment by Saklad5** [comment]
> See also [MegaMek/mekhq#4044](https://github.com/MegaMek/mekhq/issues/4044)

While similar, this issue isn't about equipment state. In MegaMek, sensors are treated separately from any specific component.

**Comment by SuperStucco** [comment]
> Yes, but configuring in the lobby would raise the potential for pre-saving sensor selection as a MUL along with other configuration. Also, I was noting one of the requests from the Discord. :-)

A more general method of associating sensor mode with chassis and/or model also has applications for Princess.  If the player thinks a certain sensor mode is good for a unit, then 'good for the goose, good for the gander'.

**Comment by Saklad5** [comment]
To some extent, yes.

I'm not proposing associations like that, though: just an ordered list of sensors, where the first one present is picked.

There are some nuances that cannot be captured by that. For instance, active probes are basically useless on the space map next to active radar, but better for spotting ground targets when flying low enough. Magscan is almost useless in urban environments, while infrared is of limited utility in hilly areas. And, of course, active probes are dead weight if they are being jammed.

Princess would be better off with that kind of logic, ultimately.

**Comment by SimonLandmine** [comment]
This has been requested again in the Discord, either as a Client Setting option or, possibly more simply, as a config option in the lobby.
Actually, as the Lobby Configuration is a different request, I'll raise that as a separate RFE.

**Comment by HammerGS** [comment]
Related https://github.com/MegaMek/megamek/issues/7362
